src/components/Product/Product.jsx

```diff
@@ -89,7 +96,35 @@ function ProductModeEdit({ id, title, detail, email, image, onDisableEditMode, o
 		setValue('email', email);
 	}, []);
 
+	const CLOUD = process.env.CLOUDINARY_CLOUD;
+
+	const PRESET = process.env.CLOUDINARY_PRESET;
+
+	const [previewImage, setPreviewImage] = useState(image);
+
+	const uploadImage = async event => {
+		try {
+			const url = `https://api.cloudinary.com/v1_1/${CLOUD}/upload`;
+			const image = event.target.files[0];
+
+			const fileData = new FormData();
+			fileData.append('file', image);
+			fileData.append('upload_preset', PRESET);
+
+			const response = await fetch(url, {
+				method: 'POST',
+				body: fileData,
+			});
+			const translation = await response.json();
+			const newImage = translation.public_id;
+			setPreviewImage(newImage);
+		} catch (error) {
+			console.error(error.message);
+		}
+	};
+
```

Comment:
Just a recommendation: If you decide to refactor at a later step, you could extract the `uploadImage` function to a separate file and reuse it in both the `Form` and `Product` component.

Reply:
Good suggestion. I will do a clean up in my next UserStory